src/array.cr

```diff
@@ -2282,14 +2520,15 @@ class Array(T)
     end
   end
 
+  # :nodoc:
   private struct FlattenHelper(T)
-    def self.flatten(ary)
+    def self.flatten(ary) : Array(T)
```

Comment:
Given this is an internal and undocumented method on an internal and undocumented type, maybe the return type restriction is of little value here?